lib/search/service.rb

```diff
@@ -114,20 +116,22 @@ def handle_429!(error)
       return unless error.status == 429
 
       StatsD.increment("#{Search::Service::STATSD_KEY_PREFIX}.exceptions", tags: ['exception:429'])
-      raise_backend_exception('SEARCH_429', self.class, error)
+      raise_backend_exception(error_code_name(error.status), self.class, error)
     end
 
     def handle_server_error!(error)
       return unless [503, 504].include?(error.status)
 
-      exceptions = {
-        503 => 'SEARCH_503',
-        504 => 'SEARCH_504'
-      }
       # Catch when the error's structure doesn't match what's usually expected.
-      message = error.body.is_a?(Hash) ? parse_messages(error).first : 'Search.gov is down'
+      message = error.body.is_a?(Hash) ? parse_messages(error).first : 'Search API is down'
       save_error_details(message)
-      raise_backend_exception(exceptions[error.status], self.class, error)
+      raise_backend_exception(error_code_name(error.status), self.class, error)
+    end
+
+    def error_code_name(error_status)
+      error_code_prefix = self.class.configuration.flipper_enabled? ? 'SEARCH_GSA' : 'SEARCH'
```

Comment:
I love this new method adding a prefix to the error code so we can tell which endpoint was hit.